app/utils.ts

```diff
@@ -0,0 +1,27 @@
+import { EXAMPLE_EIP_712_PAYLOAD } from "./constants";
+
+export function generateEip712Payload() {
+  const nonceParams = generateNonceParams();
+  const eip712Payload = { ...EXAMPLE_EIP_712_PAYLOAD };
+  eip712Payload.rpcProxySubmissionParams.message.message.nonce = nonceParams.nonce;
```

Comment:
why is this `message.message`?

Reply:
The `message` field is supposed to be an `eth_signTypedData_v4` call which has the following type:

```
{
  types: { 
    ...
  },
  domain: {
    ...
  },
  primaryType: string;
  message: {
    ....
  }
}
````

Reply:
That's... confusing 😅 , should it be named `signTypedData` instead?